Tests/CredentialsJWTTests/TestRawRouteJWT.swift

```diff
@@ -0,0 +1,289 @@
+    var jwtString2 = ""
+
+    // Key used in generation and decoding of JWT strings.
+    let key = "<PrivateKey>".data(using: .utf8)
+
+    // Sets up the raw routes for the tests.
+    var router = TestRawRouteJWT.setupRawRouter()
+
+    static var allTests : [(String, (TestRawRouteJWT) -> () throws -> Void)] {
+        return [
+        ("testDefaultTokenProfile", testDefaultTokenProfile),
+        ("testCorrectToken", testCorrectToken),
+        ("testIncorrectToken", testIncorrectToken),
+        ("testInvalidToken", testInvalidToken),
+        ("testGoogleTokenType", testGoogleTokenType),
+        ("testMissingTokenType", testMissingTokenType),
+        ("testMissingAccessToken", testMissingAccessToken)
+      ]
+
+    }
+
+    // Function that creates the logger used in debugging.
+    private static let initOnce: () = {
+        PrintLogger.use(colored: true)
+    }()
+
+    // setUp function that creates the JWT
+    override func setUp() {
+        super.setUp()
+        TestRawRouteJWT.initOnce
+
+        // User 1 JWT created.
+        performServerTest(router: router) { expectation in
+            self.performRequest(method: "post", path: "/generaterawjwt", contentType: "application/json", callback: { response in
+                do {
+                    guard let body = try response?.readString() else {
+                        XCTFail("Couldn't read response")
+                        return
+                    }
+                    self.jwtString = body
+                } catch {
+
+                }
+                expectation.fulfill()
+            }, requestModifier: { request in
+                do {
+                    try request.write(from: JSONEncoder().encode(self.testUser))
+                } catch {
+                    XCTFail("Failed to send data")
+                }
+            })
+        }
+
+        // User 2 JWT created.  Users 2 and 3 created for later tests where multiple tokens are needed.
+        performServerTest(router: router) { expectation in
+            self.performRequest(method: "post", path: "/generaterawjwt", contentType: "application/json", callback: { response in
+                do {
+                    guard let body = try response?.readString() else {
+                        XCTFail("Couldn't read response")
+                        return
+                    }
+                    do {
+                        self.jwtString2 = body
+                    }
+                } catch {
+
+                }
+                expectation.fulfill()
+            }, requestModifier: { request in
+                do {
+                    try request.write(from: JSONEncoder().encode(self.testUser2))
+                } catch {
+                    XCTFail("Failed to send data")
+                }
+            })
+        }
+    }
+
+    // Tests that the pre-constructed JWT type maps correctly to the JWT decoded from
+    // the jwtString earlier defined.
+    func testDefaultTokenProfile() {
```

Comment:
Because you are generating the jwt string rather than storing a static token, I don't think this test is necessary.